services/web/server/src/simcore_service_webserver/cli.py

```diff
@@ -1,28 +1,62 @@
-"""
- TODO: this is a placeholder for the command line interface module
+""" Application's command line .
+
+Why does this file exist, and why not put this in __main__?
+
+  You might be tempted to import things from __main__ later, but that will cause
+  problems: the code will get executed twice:
+
+  - When you run `python -msimcore_service_storage` python will execute
+    ``__main__.py`` as a script. That means there won't be any
+    ``simcore_service_storage.__main__`` in ``sys.modules``.
+  - When you import __main__ it will get executed again (as a module) because
+    there's no ``simcore_service_storage.__main__`` in ``sys.modules``.
+
 """
 import argparse
+import logging
+import sys
+
+from .cli_config import add_cli_options, config_from_options
+from .settings import CLI_DEFAULT_CONFIGFILE
+from .application import run_service
+from .settings import CONFIG_SCHEMA
 
-from . import settings
+log = logging.getLogger(__name__)
 
-def add_options(ap=None):
-    """ Creates an arguments parser and adds options to it
 
-    Returns instance of the argparse.ArgumentParser
-    """
-    # TODO: single setup per application? or use click?
-    if ap is None:
-        ap = argparse.ArgumentParser()
+def create_default_parser():
+    return argparse.ArgumentParser(description='Service to manage data storage in simcore.')
 
-    # TODO: any module could add extra options. See click!
-    ap = settings.add_cli_options(ap)
+def setup(parser):
+    """ Adds all options to a parser"""
+    parser.add_argument('names', metavar='NAME', nargs=argparse.ZERO_OR_MORE,
+                    help="A name of something.")
 
-    # TODO: Add here more options?
+    add_cli_options(parser, CLI_DEFAULT_CONFIGFILE)
 
-    return ap
+    # Add here more options ....
+
+def parse(args, parser):
+    """ Parse options and returns a configuration object """
+    if args is None:
+        args = sys.argv[1:]
 
-def parse_options(argv, ap=None):
     # ignore unknown options
-    options, _ = ap.parse_known_args(argv)
+    options, _ = parser.parse_known_args(args)
```

Comment:
why not use parse_args? you would get error messages in case the call is incorrect